New Role for Day In Life Agile Integration Workshop

satyaj opened the pull request:
<!--- Please read first:

https://github.com/redhat-cop/agnosticd/blob/development/docs/Contributing.adoc

-->
##### SUMMARY

<!--- Describe the change below, including rationale and design decisions.
The approvers and mergers shouldn't have to interpret and guess by jumping right to the code. Context helps. -->
This is a new role to deploy all the required applications for the Day In Life - Agile Integration workshop. This was based on RHMI 1.x on OCP 3.11, which is now being retired. This role replaces the previous workshop and refreshes it to latest version of OCP and Integration products.

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- New role Pull Request

##### COMPONENT NAME
<!--- Write the short name of the config, roles, task or feature below -->
ocp4-workload-dil-agile-integration

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
<!-- ansible --version -->
<!-- pip freeze -->
```paste below

```

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/remove_workload.yml:
```diff
@@ -0,0 +1,142 @@
+        api_version: apiextensions.k8s.io/v1beta1
+        kind: CustomResourceDefinition
+        label_selectors:
+        - app=strimzi
```
Please use consistent list indentaton. If `block` is indented then `label_selectors` here should also be indented. Either use extra indentation of lists everywhere or nowhere.

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/remove_workload.yml:
```diff
@@ -0,0 +1,142 @@
+        api_version: operators.coreos.com/v1alpha1
+        kind: Subscription
+        name: keycloak
+        namespace: sso
```
Please remove the block. There is no reason to have a block with one task.

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/remove_workload.yml:
```diff
@@ -0,0 +1,142 @@
+        api_version: operators.coreos.com/v1alpha1
+        kind: Subscription
+        name: threescale
+        namespace: threescale
```
Likewise here, please remove the block.

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/remove_workload.yml:
```diff
@@ -0,0 +1,142 @@
+    state: absent
+    kind: Namespace
+    name: 'user{{ item }}-fuse'
+  with_sequence: start=1 end={{ num_users }} format=%d  
```
Switch from `with_sequence` to `loop`

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/single_user.yaml:
```diff
@@ -0,0 +1,37 @@
+# - name: Evaluate Syndesis Secrets
+#   k8s:
+#     state: present
+#     resource_definition: "{{ lookup('template', 'fuse-secret.yaml.j2') }}"
```
Do not include commented code in PR

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/user_che_workspace.yaml:
```diff
@@ -0,0 +1,98 @@
+    status_code: 200
+
+
+
```
Why so many blank lines?

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/workload.yml:
```diff
@@ -0,0 +1,86 @@
+#   vars:
+#     shared_namespace: "shared"
+#     sso_project: "sso"
+#     fuse_project: "fuse"    
```
Do not include commented code.

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/workload.yml:
```diff
@@ -0,0 +1,86 @@
+    configmap_name: "extra-services"
+    apicurito_project: "apicurito"
+    microcks_project: "microcks"
+    fuse_project: "fuse"    
```
Looks like extra whitespace at the end of this line? Please look for and fix trailing whitespace errors.

jkupferer replied:
Still some trailing space here.

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/templates/user_registration.yaml.j2:
```diff
@@ -0,0 +1,237 @@
+          - name: LAB_DURATION_HOURS
+            value: '2weeks'
+          - name: LAB_USER_PASSWORD
+            value: '{{ ocp4_workload_dil_agile_integration_openshift_user_password }}'
```
For quoted strings like this it would be safer to use `to_json` to quote rather than `'`. For example:
```
            value: {{ ocp4_workload_dil_agile_integration_openshift_user_password | to_json }}
```

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/provision_gogs.yaml:
```diff
@@ -0,0 +1,132 @@
+    status_code: 200
+  register: create_tenant_result
+  ignore_errors: true
+  with_sequence: start=0 end={{ num_users }} format=user%d
```
Please switch to `loop`

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/provision_online.yaml:
```diff
@@ -0,0 +1,63 @@
+    state: present
+    namespace: '{{ operators_project }}'
+    src: '{{ item }}'
+  with_items: "{{ install_files.files | map(attribute='path') | list }}"
```
Switch to `loop`

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/provision_sso.yaml:
```diff
@@ -0,0 +1,57 @@
+  include_tasks: user_sso_realm.yaml
+  with_sequence: start=1 end={{ num_users }} 
+  when:
+    - num_users|int > 0
```
Switch to `loop` and set a `loop_var` with `loop_control`

jkupferer replied:
I recommend setting `loop_var` when calling `include_tasks` in `loop`. It is dangerous not to and can lead to some really surprising errors if the included tasks file has any `loop` blocks if we don't have this.

```
- name: Create User realms
  include_tasks: user_sso_realm.yaml
  loop: "{{ range(1, num_users|int + 1, 1)|list }}"
  loop_control:
    loop_var: __user_number
  when:
    - num_users|int > 0 
```

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/provision_threescale.yaml:
```diff
@@ -0,0 +1,74 @@
+  include_tasks: user_tenant.yaml
+  with_sequence: start=1 end={{ num_users }} 
+  when:
+    - num_users|int > 0
```
Switch to `loop` and use a `loop_var` in `loop_control`

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/provision_webapp.yaml:
```diff
@@ -0,0 +1,70 @@
+    state: present
+    namespace: '{{ webapp_namespace }}'
+    src: '{{ webapp_operator_tmp }}/tutorial-web-app-operator-{{ ocp4_workload_dil_agile_integration_webapp_operator_release_tag }}/deploy/{{ item }}'
+  with_items: "{{ ocp4_workload_dil_agile_integration_webapp_operator_resource_items }}"
```
Switch to `loop`

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/remove_workload.yml:
```diff
@@ -0,0 +1,142 @@
+    state: absent
+    kind: Namespace
+    name: 'user{{ item }}'
+  with_sequence: start=1 end={{ num_users }} format=%d
```
Switch to `loop`

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/remove_workload.yml:
```diff
@@ -0,0 +1,142 @@
+    state: absent
+    kind: Namespace
+    name: 'user{{ item }}-che'
+  with_sequence: start=1 end={{ num_users }} format=%d
```
Switch to `loop`

satyaj commented:
@jkupferer thanks for the detailed feedback. I have fixed the issues raised here and re-tested the workload.
Please review and merge. Thanks, Satya

jkupferer reviewed (changes requested):
Looks like a few items left to address.

satyaj commented:
Hi @jkupferer ,
Thanks. I fixed those 2 issues raised as well. Also removed AMQ Online as that is not being used in current set of labs.

Thanks
Satya

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/provision_che.yaml:
```diff
@@ -0,0 +1,122 @@
+    - [ 'client_id', 'admin-cli' ]
+    - [ 'username', 'admin' ]
+    - [ 'password', '{{ key_cloak_admin_password }}' ]
+    - [ 'grant_type', 'password' ]
```
The `uri` `body` supports passing a dict, probably easier to read here.
```
    body:
      client_id: admin-cli
      username: admin
      password: '{{ key_cloak_admin_password }}'
      grant_type: password
```

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/provision_che.yaml:
```diff
@@ -0,0 +1,122 @@
+    - [ 'client_id', 'admin-cli' ]
+    - [ 'username', 'admin' ]
+    - [ 'password', '{{ key_cloak_admin_password }}' ]
+    - [ 'grant_type', 'password' ]
```
Same here, consider using a dictionary.

jkupferer commented on ansible/roles/ocp4-workload-dil-agile-integration/tasks/provision_gogs.yaml:
```diff
@@ -0,0 +1,143 @@
+    url: 'http://gogs.{{ route_subdomain }}/api/v1/repos/migrate'
+    method: POST
+    body: 
+      '{"clone_addr": "https://github.com/weimeilin79/dayinthelife-import.git", "uid": {{ item | int }},"repo_name": "dayintheliferepo"}'
```
AgnosticD code should not depend on any repository owned by an individual. The repository should be managed by an organization such as https://github.com/redhat-cop/

I see that we can't use a dict here because of the nede to pass the `uid` as an int.
